### PR TITLE
fix length bug - doubly incremented n

### PR DESCRIPTION
**Pull request by petabytestorage**
incremented in WriteTo and before. two places, same pointer address so same int incremented doubly incorrectly.

### CODE REVIEW

**Review by melekes — APPROVED** [review]

File: int_test.go
Comment (odeke-em) [inline int_test.go:11]: @petabytestorage could we enumerate the same test for the respective *int* Writers and the expected values? That'll help us catch and prevent any regressions.

**Review by odeke-em — CHANGES_REQUESTED** [review]
Just a request for more test enumerations for the other functions and bug fixes.

File: int_test.go
Comment (petabytestorage) [reply int_test.go:11]: i'm rewriting all this code in a better interface that eliminates this so i will leave it up to @ebuchman and @jaekwon how much effort that want to put into fixing this top-level int.go (depends how much it's used i guess in other places?) versus fixing the new one in nowriter/tmlegacy/... and indeed we are moving away from that interface and style so all these bugs will go away as we move into the `[]byte` format that @jaekwon has supported for some time. so i think fastest is just finish up the new good interfaces, add simple wrapper to support buf accumulation (i will be adding that soon) then we can move everything over to new safer interface and have all these bugs gone because we no longer manipulate int n byte count directly because we no longer use old legacy interface.  therefore my suggestion is merge my other pr and lets get finishing. https://github.com/tendermint/go-wire/pull/51

Comment (odeke-em) [reply int_test.go:11]: @petabytestorage sure, but the problem though is that we had that code in there and no one noticed those bugs, I believe enumerating those tests is a copy and paste and changing the sizes. If you'd like I could send a follow up PR to implement them. I agree, finish up as much.

Comment (petabytestorage) [reply int_test.go:11]: i accept that offer!  :ok_woman: ty @odeke-em

Comment (odeke-em) [reply int_test.go:11]: Cheers :), please ping me once this PR has been merged and I'll send them along.

Comment (petabytestorage) [reply int_test.go:11]: i think if you make a PR to be merged into this PR then i merge it it would clear your old blocking review and then you can approve. so save a step or two not sure tho about this github workflow gui thing.

**Review by ebuchman — APPROVED** [review]
OMG lol. Thanks Rudi.

File: int_test.go
Comment (petabytestorage) [reply int_test.go:11]: it's merge @odeke-em

**Pull request merged**
